examples/host_camera_server.cc

```diff
@@ -70,7 +71,7 @@ typedef struct stream_ctx_t
 
 stream_ctx_t *stream_ctx;
 
-char device_index[] = "/dev/video0";
+char device_index[MAX_DEV_NAME_LEN] = "/dev/video0";
```

Comment:
Project declares usage of quite modern c++ standard, i.e. c++17:
https://github.com/projectceladon/libvhal-client/blob/d16a27443a1336f581a376b708139730d5b870ca/CMakeLists.txt#L22

Why do you keep struggling with C style "char" array and "strncpy" if you can simplify all out and avoid any static analysis issues around that (and there are many!):
```
std::string dev_video = "/dev/video0"
<...>
void get_all_dev_nodes()
{
    for(int devId = 0; devId < 64; devId++) {
        struct v4l2_capability video_cap{};
        std::string dev = "/dev/video" + std::to_string(devId);
        int fd = open(dev.c_str(), O_RDONLY); 
        if(fd == -1) {
            continue;
	}
        <...>
        if(ioctl(fd, VIDIOC_QUERYCAP, &video_cap) == -1)
            cout <<"cam_info: Can't get capabilities\n";
        else {
	    if(!strcmp((const char*)video_cap.driver, "v4l2 loopback"))
	        dev_video = std::move(dev);
        }
        close(fd);
    }
}
```

Mind also:
* `dev_video` name instead of `device_index` - you store full device path, not index, so name it accordingly
* Usage of scope variables in `for` loop instead of declaring out of the loop - this avoids usage of variables set on the previous loop iteration
* Initialization of `video_cap` with default initializer `{}`

Reply:
Used strings instead of c style declarations.
Updated the host_camera_server.cc by removing unused variables and tab spaces.
Addressed the review comments. Please review